core/src/main/scala/chisel3/Bits.scala

```diff
@@ -25,7 +25,9 @@ private[chisel3] sealed trait ToBoolable extends Element {
     *
     * @note The width must be known and equal to 1
     */
-  final def asBool(): Bool = macro SourceInfoWhiteboxTransform.noArg
+  final def asBool: Bool = macro SourceInfoWhiteboxTransform.noArg
+  @deprecated("Calling a nullary function with an empty argument list will be removed by official release", "chisel3")
```

Comment:
```suggestion
  @deprecated("Calling a nullary function with an empty argument list is not valid in Scala3. Use .asBool without parentheses", "3.5")
```

I think a good deprecation message should as clearly as possible what you should do if you get this deprecation message. And I thought the second argument is the version in which this was first deprecated?

Reply:
Yes you should always tell the user what to do. I don't think it's necessary to say "Use `.asBool` without parentheses", you could just say "Use form without parentheses" or something, this would make it easier to copy-paste the deprecation method throughout.